### DISCUSSION TITLE
How does Sequoia change the future of Rectangle?

### DISCUSSION THREAD

**Discussion by euphemism** [rxhanson/Rectangle#1406] [General]
Hey there, watching the WWDC24 stream, and the new MacOS version now has much of the Rectangle functionality built in.

**Comment by TouhouEngie** [comment]
Some people might continue using Rectangle for reasons similar to people using PowerToys’ built-in window snapping—both Rectangle and PowerToys offer more customization than stock offerings.

With macOS Sequoia, I assume they won’t allow power users to change the shortcuts without third-party tools.

Edit: This has been proved! Also many people are saying that stock snapping is 2x2 only

**Reply by bernaferrari** [reply]
And Sequoia has a 1-2 second delay when dragging to the top which is awful. It can't even get the basics right. Rectangle is still better.

**Comment by rxhanson** [comment]
I'm not sure exactly what things will look like going forward, but Rectangle will continue to be supported. I have no plans to sunset it :)

**Reply by Aaron-Rumpler** [reply]
[By the looks of it](https://www.youtube.com/watch?v=5OPA1Xg4CXg), the window snapping in macOS Sequoia is quite basic (halves and quarters only).

**Reply by bigplayer-ai** [reply]
> I'm not sure exactly what things will look like going forward, but Rectangle will continue to be supported. I have no plans to sunset it at this time. :)

is there any new api that utilises the new tiling/window snipping? or is it only the same accessibility api?

**Reply by rxhanson** [reply]
As far as I know, there is no new API.

**Comment by bernaferrari** [comment]
I updated to Sequoia and it is waaaay worse than Rectangle. Sure, the animation is nice (if there is an API Rectangle could adopt it too), but you can't tile by dragging to the top, there is a 1-2 second delay which gets me crazy, in order to not interfere with moving to other spaces.

I'll stay with Rectangle. I already disabled the macOS option.

**Reply by mimoo** [reply]
I couldn't figure out how to split your screen in two halves, it keeps a large void in the middle of the screen for some reason, so now I'm back at using rectangle

**Comment by injust** [comment]
Feature request: Could we get Sequoia's "Hold ⌥ key while dragging windows to tile" functionality in Rectangle?

Not sure how it should look in Rectangle though, since Rectangle's snap-by-dragging snaps to thirds on the bottom edge.

**Reply by rxhanson** [reply]
Here's the Terminal command to configure this in Rectangle:
https://github.com/rxhanson/Rectangle/blob/main/TerminalCommands.md#only-allow-drag-to-snap-when-modifier-keys-are-pressed

To set it to option, you would use 
```
defaults write com.knollsoft.Rectangle snapModifiers -int 524288
```

Restart the app after executing it.

**Reply by injust** [reply]
I gave that a try, but it's not the same behaviour. That setting enables Rectangle's snap-at-edge functionality _only_ when ⌥ is pressed.

Sequoia's functionality keeps snap-at-edge, and (when ⌥ is pressed) additionally lets you snap a window without the cursor being at the screen edge.

**Reply by rxhanson** [reply]
Ah, good to know. I'll have to test that out again.

**Comment by EthanJStark** [comment]
Super grateful for Rectangle, it's legit one of my favorite apps.

Fully agree with the discussion here, it's definitely not obsolete. For me, the fact that Sequoia does not appear to permit customizing shortcuts is a deal breaker.

❓: When I first upgraded, I was presented with a window (maybe from MacOS? Maybe from Rectable? Idk) asking me to pick Sequoia OR Rectangle key bindings . . . I picked Sequoia, curious to try it out.

Anyone have any clues what this dialogue was? Rectangle continues to work great for me.

Thanks again, @rxhanson !!

**Reply by rxhanson** [reply]
Always good to hear about sticking with Rectangle!

The dialog is within Rectangle. It's intended to prevent the scenario where you have both Rectangle & macOS operating on the screen edge drag at the same time.

**Reply by jmb** [reply]
How do I get that question again? I'm not sure which one I clicked - I think I clicked use MacOS but maybe use Rectangle is better... 😄

**Reply by bernaferrari** [reply]
You just disable in macos menu and enable rectangle

**Reply by rxhanson** [reply]
The macOS setting is in System Settings > Desktop & Dock > Windows, and the Rectangle setting is in the second tab of the settings window.
